sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -314,8 +314,8 @@ case class AlterTableChangeColumnCommand(
     val resolver = sparkSession.sessionState.conf.resolver
     DDLUtils.verifyAlterTableType(catalog, table, isView = false)
 
-    // Find the origin column from schema by column name.
-    val originColumn = findColumnByName(table.schema, columnName, resolver)
+    // Find the origin column from dataSchema by column name.
+    val originColumn = findColumnByName(table.dataSchema, columnName, resolver)
```

Comment:
Only data schema? How about partitioning schema?

Reply:
As described in the comment, we only support altering non-partition columns:
```
 A command to change the column for a table, only support changing the comment of a non-partition
 column for now.
```

Reply:
Do we have a negative test case to cover that?